### PR TITLE
Support all fields reference from outer scope relation

### PR DESCRIPTION
**Pull request by kasiafi**

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/sql/query/TestSelectAll.java [hunk 1]
```diff
@@ -25,6 +25,8 @@
 {
     private QueryAssertions assertions;
 
+    private static final String UNSUPPORTED_CORRELATED_SUBQUERY_ERROR_MSG = ".*: Given correlated subquery is not supported";
```
Comment (martint) [inline presto-main/src/test/java/io/prestosql/sql/query/TestSelectAll.java]: I'd just call this `UNSUPPORTED_DECORRELATION_MESSAGE`

File: presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java [hunk 2]
```diff
@@ -1035,6 +1035,41 @@ public void testSelectAllFromTable()
         assertQuery("SELECT y FROM (SELECT r.* AS (x, y, z) FROM region r ORDER BY y DESC LIMIT 2)", "VALUES 'MIDDLE EAST', 'EUROPE'");
     }
 
+    @Test
+    public void testSelectAllFromOuterScopeTable()
```
Comment (martint) [inline presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java]: Do we have any tests where the alias shadows an alias or table name from an outer scope? Maybe also one-level removed, where the alias binds against the outer scope, but there's another alias with the same name one level further?

File: presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java [hunk 3]
```diff
@@ -1980,11 +1986,20 @@ private void analyzeAllColumnsFromTable(
...
-                analyzeExpression(fieldReference, scope);
-                outputExpressionBuilder.add(fieldReference);
-                selectExpressionBuilder.add(new SelectExpression(fieldReference, Optional.empty()));
+                Expression fieldExpression;
+                if (local) {
+                    fieldExpression = new FieldReference(relationType.indexOf(field));
+                }
+                else {
+                    if (!field.getName().isPresent()) {
+                        throw semanticException(NOT_SUPPORTED, node.getSelect(), "SELECT * from outer scope table not supported with anonymous columns");
```
[the remark this answers is not in the document]
Comment (kasiafi) [reply presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java]: It's here in `TestAnalyzer`, line 290:
```
// reference to outer scope relation with anonymous field
        assertFails("SELECT (SELECT outer_relation.* FROM (VALUES 1) inner_relation) FROM (values 2) outer_relation")
                .hasErrorCode(NOT_SUPPORTED);
```

File: presto-main/src/test/java/io/prestosql/sql/query/TestSelectAll.java [hunk 4]
```diff
@@ -214,6 +217,7 @@ public void testSelectAllFromOuterScopeTable()
         assertions.assertFails("SELECT * FROM (VALUES 0, 1) t(a), LATERAL (SELECT t.* WHERE t.a = 0)", UNSUPPORTED_DECORRELATION_MESSAGE);
         // FROM in lateral relation
         assertions.assertQuery("SELECT * FROM (VALUES 0, 1) t(a), LATERAL (SELECT t.* FROM (VALUES 1))", "VALUES (0, 0), (1, 1)");
+        assertions.assertQuery("SELECT t.* FROM (VALUES 0, 1) t(a), LATERAL (SELECT t.*) t\n", "VALUES (0, 0), (1, 1)");
```
Comment (martint) [inline presto-main/src/test/java/io/prestosql/sql/query/TestSelectAll.java]: Extra unnecessary `\n`

**Comment by martint** [comment]
Merged, thanks!

**Pull request closed without merging**